File: challenges/sum-pairs/python/sum_pairs.py
def sum_pairs(array, test_number):
    sum_pair_list = list()  # list to store sum pairs

    for num in array:  # still number
        for i in array:  # interate through list checking num + i
            if num + i == test_number:  # test sum
                if [i, num] in sum_pair_list:  # checks if pair is in list
                    return sum_pair_list
                else:
                    sum_pair_list.append([num, i])

    if len(sum_pair_list) == 0:  # check if no pairs were found
        return 'unable to find pairs'

    return sum_pair_list

File: challenges/sum-pairs/js/sumPairs.js
// const sumPairs = (array, number) => {
//     const sum_pairs_list = [] // holds sum pairs

//     for (let i = 0; i < array.length - 1; i++) { // fixed number
//         for (let j = i + 1; j < array.length; j++) { // check number
//             if (array[i] + array[j] === number) {
//                 sum_pairs_list.push([array[i], array[j]])
//             }
//         }
//     }
//     if (sum_pairs_list.length === 0) {
//         return 'unable to find pairs'
//     }
//     return sum_pairs_list
// }

const sumPairs = (array, number) => {
    const sum_pairs_list = [] // holds sum pairs
    const sorted_pairs = array.sort()
    let right = sorted_pairs.length - 1
    let left = 0

    while (left < right) {
        let sum = sorted_pairs[right] + sorted_pairs[left]
        if (sum === number) {
            sum_pairs_list.push([sorted_pairs[left], sorted_pairs[right]])
            right--
        } else if (sum < number) {
            left++
        } else {
            right--
        }
    }
    if (sum_pairs_list.length) {
        return sum_pairs_list
    } else {
        return 'unable to find pairs'
    }
}


module.exports = { sumPairs }